ui/src/components/comment-node.tsx

```diff
@@ -852,12 +884,20 @@ export class CommentNode extends Component<CommentNodeProps, CommentNodeState> {
     i.setState(i.state);
   }
 
-  handleDeleteClick(i: CommentNode) {
+  displayDeletionOrRestorationConfirmation(showConfirmation: boolean) {
+    return function (i: CommentNode) {
+      i.state.showDeletionOrRestorationConfirmation = showConfirmation;
+      i.setState(i.state);
+    };
+  }
+
+  handleDeletionOrRestoration(i: CommentNode) {
     let deleteForm: DeleteCommentForm = {
       edit_id: i.props.node.comment.id,
       deleted: !i.props.node.comment.deleted,
       auth: null,
     };
+    i.state.showDeletionOrRestorationConfirmation = false;
```

Comment:
This also needs a setState here. 

Just look at `handleModRemoveSubmit` for an example.